source/baseObject.py

```diff
@@ -128,6 +128,22 @@ def invalidateCaches(cls):
 		for instance in cls.__instances.keys():
 			instance.invalidateCache()
 
+class ScriptableType(AutoPropertyType):
+	"""A metaclass used for collecting and caching gestures on a ScriptableObject"""
+
+	def __new__(meta, name, bases, dict):
+		gestures = {}
+		cls = super(ScriptableType, meta).__new__(meta, name, bases, dict)
+		for name, script in dict.iteritems():
+			if not name.startswith('script_'):
+				continue

```

Comment:
I wonder if this should log a warning? It could be quite frustrating trying to track down something like this.

Reply:
I guess I didn't understand this correctly. I was trying to suggest a way to catch where someone has incorrectly placed the `@script` decorator on a method that does not start with `script_`

Reply:
Hmm that's an interesting idea. We could easily make the script decorator raise a ValueError if the provided script has a __name__ that does not start with script. We should keep in mind that we can not catch such errors, so it would drive the complete module unusable.

AN alternative approach could be, make the script decorator set a _isDecorated boolean on the script, and check for that variable on every item in the class' dictionary.